### PR TITLE
Users list

### PR DESCRIPTION
**Pull request by fede1608**
## Summary
- GET /users endpoint implemented with pagination
- Authorization using tokens with express-jwt
- Tests
- Docs
## Trello Card
[https://trello.com/c/GaqnU9Dh/4-lista-de-usuarios](https://trello.com/c/GaqnU9Dh/4-lista-de-usuarios)

### CODE REVIEW

File: app/routes.js [hunk 1]
```diff
@@ -49,4 +50,6 @@ exports.init = app => {
     ],
     userController.userNewSession
   );
+
+  app.get('/users', [jwt({ secret: config.secret })], userController.usersList);
```
Comment (mpizzagalli) [inline app/routes.js:54]: I would personally prefer to move this to another middleware file and abstract which package or logic I use there and here only call that customizable middleware.
But this is correct.

File: app/controllers/users.js [hunk 2]
```diff
@@ -53,5 +54,18 @@ module.exports = {
         logger.error(`DB: ${error.errors[0]}`);
         res.status(500).json({ error: error.errors[0].message });
       });
+  },
+  usersList(req, res, next) {
+    let page = parseInt(req.query.page) || 1;
+    page = page > 0 ? page : 1;
+    const pageSize = 10;
```
Comment (mpizzagalli) [inline app/controllers/users.js]: the page size should also be a param, with a given default

File: app/controllers/users.js [hunk 3]
```diff
@@ -53,5 +54,18 @@ module.exports = {
...
+    User.findAll({ offset: pageSize * (page - 1), limit: pageSize })
```
Comment (mpizzagalli) [inline app/controllers/users.js]: As a good practice, is nice to always return the promises. Here in the controller or any middleware it doesn't makes much difference but once you start to delegate and modularize into another functions and another files, you will be adding a lot of returns and potentialy missing some of them and having some headaches.

File: app/controllers/users.js [hunk 4]
```diff
@@ -53,5 +54,18 @@ module.exports = {
...
+      .then(users => {
+        res.status(200).json({ users, page });
```
Comment (mpizzagalli) [inline app/controllers/users.js]: We should also send some information about the total pages or the total amount of elements, so the front or mobile can display de UI correctly.

File: test/users.js [hunk 5]
```diff
@@ -250,3 +250,51 @@ describe('/users/sessions POST', () => {
...
+  });
+
+  it('should fail if invalid token', done => {
+    chai
+      .request(server)
+      .get('/users')
+      .set('authorization', `Bearer 1234123`)
+      .end((err, res) => {
+        expect(err).not.to.be.null;
+        expect(res).to.have.status(500);
```
Comment (mpizzagalli) [inline test/users.js:285]: If invalid or missing token the `express-jwt` package is responding 500?? I don't like that, 500 would mean that there is something wrong server-side, but clearly is something wrong with the client request, so it should be a 4xx.

**Review by mpizzagalli — CHANGES_REQUESTED** [review]

File: app/routes.js (hunk 1 above)
Comment (fede1608) [reply app/routes.js:54]: Done that in #13

File: app/controllers/users.js (hunk 2 above)
Comment (fede1608) [reply app/controllers/users.js]: Fixed

File: app/controllers/users.js (hunk 4 above)
Comment (fede1608) [reply app/controllers/users.js]: Done

File: test/users.js (hunk 5 above)
Comment (fede1608) [reply test/users.js:285]: Fixed in #13

**Review by mpizzagalli — APPROVED** [review]

**Pull request closed without merging**